src/main/resources/lib/ssb/utils/customHooks/calculatorHooks.tsx

```diff
@@ -0,0 +1,342 @@
+    : ''
+  const [startValue, setStartValue] = useState({
+    error: false,
+    errorMsg: calculatorValidateAmountNumber,
+    value: '',
+  })
+  const [startMonth, setStartMonth] = useState<CalculatorState>({
+    error: false,
+    errorMsg: lastNumberText,
+    value: defaultMonthValue,
+  })
+
+  const [startYear, setStartYear] = useState<CalculatorState>({
+    error: false,
+    errorMsg: validYearErrorMsg,
+    value: '',
+  })
+
+  const [endMonth, setEndMonth] = useState<CalculatorState>({
+    error: false,
+    errorMsg: lastNumberText,
+    value: defaultMonthValue,
+  })
+
+  const [endYear, setEndYear] = useState<CalculatorState>({
+    error: false,
+    errorMsg: validYearErrorMsg,
+    value: '',
+  })
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
+  const [loading, setLoading] = useState(false)
+  const [change, setChange] = useState<string | null>(null)
+  const [endValue, setEndValue] = useState<null | number>(null)
+  const [startPeriod, setStartPeriod] = useState<string | null>(null)
+  const [endPeriod, setEndPeriod] = useState<string | null>(null)
+  const [startValueResult, setStartValueResult] = useState<string | null>(null)
+  const [startIndex, setStartIndex] = useState<number | null>(null)
+  const [endIndex, setEndIndex] = useState<number | null>(null)
+
+  const scrollAnchor = useRef<HTMLDivElement>(null)
+  const onSubmitBtnElement = useRef<HTMLButtonElement>(null)
+
+  const yearRegexp = /^[1-9]\d{3}$/g
+
+  function isStartValueValid(value?: string) {
+    const startVal = value || startValue.value
+    const testStartValue = startVal.match(/^-?\d+(?:[.,]\d*)?$/g)
+    const isNumber = testStartValue && testStartValue.length === 1
+    return !(!isNumber || isNaN(parseFloat(startVal)))
+  }
+
+  function isStartYearValid(value?: string) {
+    const startYearValue = value || (startYear.value as string)
+    const testStartYear = startYearValue.match(yearRegexp)
+    const isStartYearValid = testStartYear && testStartYear.length === 1
+    const intStartYear = parseInt(startYearValue)
+    return !(
+      !isStartYearValid ||
+      isNaN(intStartYear) ||
+      intStartYear < validMinYear ||
+      intStartYear > Number(validMaxYear)
+    )
+  }
+
+  function isEndYearValid(value?: string) {
+    const endYearValue = value || (endYear.value as string)
+    const testEndYear = endYearValue.match(yearRegexp)
+    const isEndYearValid = testEndYear && testEndYear.length === 1
+    const intEndYear = parseInt(endYearValue)
+    return !(!isEndYearValid || isNaN(intEndYear) || intEndYear < validMinYear || intEndYear > Number(validMaxYear))
+  }
+
+  function isStartMonthValid(value?: string) {
+    const startMonthValue = value || (startMonth.value as DropdownItem).id
+    const isQuartalPeriod = startMonthValue !== '' && isNaN(Number(startMonthValue))
+    const startMonthOrQuartalPeriodMonthValue = isQuartalPeriod
```

Comment:
Hadde kanskje vært mer logisk å ha egne funksjoner for isStartQuartalValid og isEndQuartalValid siden det er forskjellige tidsenheter.